### ISSUE TITLE
Would be nice to be able to merge metadata columns when doing an analysis

### ISSUE THREAD

**Issue by clozupone** [biocore/qiita#1446]
I remember there being functionality in the past (I think in the old QIIME database) where one could merge info in two different metadata categories "on the fly" when doing an analysis. I thought that this functionality was pretty convenient/useful.

**Comment by antgonza** [comment]
Agree but I think this would be nicer on the visualization per se. For example, within Emperor. As predefining the categories that you want to merge means that you already know what you are looking for, which is not generally the case. What do you think @clozupone?

**Comment by clozupone** [comment]
Yes, I agree - I think adding to emperor would be better. I can add as a feature request there instead….
Thanks!
Cathy

On Aug 26, 2015, at 11:10 AM, Antonio Gonzalez <notifications@github.com<mailto:notifications@github.com>>
 wrote:


Agree but I think this would be nicer on the visualization per se. For example, within Emperor. As predefining the categories that you want to merge means that you already know what you are looking for, which is not generally the case. What do you think @clozupone<https://github.com/clozupone>?

**Comment by rob-knight** [comment]
If adding to emperor, it might be worth considering how to export the new categories from emperor so that they can be used for all the other analyses in qiime/qiita after they’ve been manually refined?

**Comment by antgonza** [comment]
I'm not sure we want to store the merged column in the metadata DB as it will create duplicated fields. However, we could find a way to save those combinations in the analysis. One way that comes to mind is the idea of saving Emperor status via the URL, which you then could save in your analysis and put a description. Then you could share it with other users, as described here: https://github.com/biocore/qiita/issues/1444. What do you think?

**Comment by josenavas** [comment]
@antgonza we might be able to put that column back in the mapping file for the analysis, but do not modify the templates of the actual studies, as those columns might be the result of merging 2 columns from 2 different studies.

**Comment by ElDeveloper** [comment]
I agree, this feature would fit great in emperor and being able to export newly created mapping files will also be tremendously helpful! I also think that Qiita should be capable of creating new columns in the mapping file mainly to make it easier to extend the usability of other tools/scripts that are not as interactive, think of statistical tests, taxa summaries, etc.

**Comment by adswafford** [comment]
@ElDeveloper This is the third or fourth issue I've found related to exporting metadata from Emperor, can we do a status update on whether that idea was abandoned for lack of time or lack of merit?

**Comment by antgonza** [comment]
Emperor is still in development (Q2 version), slowly but still moving ...

**Comment by ElDeveloper** [comment]
Right, and for this the main thing that Emperor would need is an 
endpoint in the server that it can ping when such an action is 
triggered. We also need a way to select samples in a lasso-kinda 
fashion, which happens to be kinda complicated in WebGL.

**Comment by antgonza** [comment]
@adswafford & @ElDeveloper, should we move this issues to https://github.com/biocore/emperor so they are not buried in here ...

**Comment by ElDeveloper** [comment]
+1
